lib/puppet/provider/exec/posix.rb

```diff
@@ -2,14 +2,18 @@
 
 Puppet::Type.type(:exec).provide :posix, :parent => Puppet::Provider::Exec do
   has_feature :umask
+  has_feature :supports_pass_through_shell
+
```

Comment:
I'm curious why you're using features instead of defining a method on the base `Puppet::Provider::Exec` provider?

Reply:
No particular reason, I was trying to confine the param only to the `posix` implementation.